rdrf/rdrf/auth/views.py

```diff
@@ -6,32 +6,65 @@
 from django.contrib.auth import get_user_model
 
 from django.contrib.auth.decorators import login_required
+from django.contrib.auth.signals import user_logged_in
 from django.contrib.auth.forms import SetPasswordForm
 
 from django.contrib.auth.tokens import default_token_generator
 from django.core.exceptions import ObjectDoesNotExist
+from django.dispatch import receiver
 from django.http import HttpResponseRedirect
 from django.shortcuts import resolve_url
 from django.template.response import TemplateResponse
 from django.urls import reverse
 from django.utils.encoding import force_text
+from django.utils.safestring import mark_safe
 from django.utils.http import urlsafe_base64_decode
 from django.utils.translation import ugettext as _
 from django.views.decorators.cache import never_cache
 from django.views.decorators.debug import sensitive_post_parameters
 
+from two_factor import views as tfv
+from two_factor.utils import default_device
+
 from useraudit.models import UserDeactivation
 from useraudit.password_expiry import is_password_expired
 
 from registry.patients.models import Patient, ParentGuardian
 
-from . import can_user_self_unlock
+from . import can_user_self_unlock, is_user_privileged
 from .forms import UserVerificationForm, RDRFSetPasswordForm, ReactivateAccountForm
 
 
 logger = logging.getLogger(__name__)
 
 
+@receiver(user_logged_in)
+def user_login_callback(sender, request=None, user=None, **kwargs):
+    if is_user_privileged(user) and default_device(user) is None:
+        link = ('<a href="%(url)s" class="alert-link">' + _('click here') + '</a>') % {'url': reverse('two_factor:setup')}
+        msg = mark_safe(
+                _('We strongly recommend that you protect your account with Two-Factor authentication. '
```

Comment:
So this message will always appear as a nag ?

Reply:
Yes, but only for "system users".

Reply:
I found that if as an admin I made a user use 2fa and then logged in as that user I get the 2fa setup screen as expected, but after I set up the 2fa ,  am still greeted by that 2fa recommendation message ( despite having just set it up)

Reply:
Also after being "forced" to setup 2fa by an admin, the disable 2fa option still exists in the drop down, which I can then click. Then when I log back in I'm forced to set it up again.

So if 2fa is being made mandatory ( via the admin setting it as required in the admin interface), shouldn't the cancel option simply not be presented to the user at all in this case?